src/shared-validator.cc

```diff
@@ -518,6 +518,18 @@ Result SharedValidator::CheckAlign(const Location& loc,
   return Result::Ok;
 }
 
+Result SharedValidator::CheckOffset(const Location& loc,
+                                    Address offset,
+                                    const Limits& limits) {
+  if (limits.is_64 || (offset <= UINT32_MAX)) {
+    return Result::Ok;
+  }
+
+  PrintError(loc, "offset must be less than or equal to 0xffffffff");
```

Comment:
I has a slight preference for flipping the logic to we error out early on wasm32 and return `Ok` otherwise.

Reply:
Done.